src/chart_process/types.rs

```diff
@@ -198,11 +325,71 @@ impl From<BmpId> for usize {
     }
 }
 
+/// Chart event unique identifier wrapper type
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ChartEventId(pub usize);
+
+impl ChartEventId {
+    /// Create a new ChartEventId
+    #[must_use]
+    pub const fn new(id: usize) -> Self {
+        Self(id)
+    }
+
+    /// Get the internal usize value
+    #[must_use]
+    pub const fn value(self) -> usize {
+        self.0
+    }
+}
+
+impl From<usize> for ChartEventId {
+    fn from(value: usize) -> Self {
+        Self(value)
+    }
+}
+
+impl From<ChartEventId> for usize {
+    fn from(id: ChartEventId) -> Self {
+        id.0
+    }
+}
+
+/// Generator for sequential `ChartEventId`s
+#[derive(Debug, Clone, Default)]
+pub struct ChartEventIdGenerator {
+    next: usize,
+}
+
+impl ChartEventIdGenerator {
+    /// Create a new generator starting from `start`
+    #[must_use]
+    pub const fn new(start: usize) -> Self {
+        Self { next: start }
+    }
+
+    /// Allocate and return the next `ChartEventId`
+    #[must_use]
+    pub const fn next_id(&mut self) -> ChartEventId {
+        let id = ChartEventId(self.next);
+        self.next += 1;
+        id
+    }
+
+    /// Return the next `ChartEventId` that will be used
+    #[must_use]
+    pub const fn peek_next(&self) -> ChartEventId {
+        ChartEventId::new(self.next)
+    }
+}
+
 /// Timeline event and position wrapper type.
 ///
 /// Represents an event in chart playback and its position on the timeline.
 #[derive(Debug, Clone)]
 pub struct ChartEventWithPosition {
```

Comment:
Maybe we should provide `PartialEq`/`Eq` implementations for `ChartEventWithPosition`.